File: problems/LeetCode/TopInterview150/longestCommonPrefix.ts
function longestCommonPrefix(strs: string[]): string {
  let ans = '';

  for (let i = 0; i < strs[0].length; i++) {
    let isCommon = true;
    const char = strs[0][i];

    for (let j = 1; j < strs.length; j++) {
      if (!strs[j][i] || strs[j][i] !== char) {
        isCommon = false;
        return ans;
      }
    }

    if (isCommon) {
      ans += char;
    }
  }

  return ans;
}

File: problems/LeetCode/TopInterview150/longestCommonPrefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:

        if not strs:
            return ""

        for i, char in enumerate(strs[0]):
            for word in strs[1:]:
                if i >= len(word) or char != word[i]:
                    return strs[0][:i]

        return strs[0]
    

# Time: O(m*n)
# Space: O(1)